In [1]:
import pandas as pd

In [2]:
# 1. Read this dataset in pandas , mysql and mongodb

In [3]:
df = pd.read_csv("FitBit Data.csv")

In [4]:
df

Id ActivityDate  TotalSteps  TotalDistance  TrackerDistance  \
0    1503960366    3/25/2016       11004       7.110000         7.110000   
1    1503960366    3/26/2016       17609      11.550000        11.550000   
2    1503960366    3/27/2016       12736       8.530000         8.530000   
3    1503960366    3/28/2016       13231       8.930000         8.930000   
4    1503960366    3/29/2016       12041       7.850000         7.850000   
..          ...          ...         ...            ...              ...   
452  8877689391   04-08-2016       23014      20.389999        20.389999   
453  8877689391   04-09-2016       16470       8.070000         8.070000   
454  8877689391   04-10-2016       28497      27.530001        27.530001   
455  8877689391   04-11-2016       10622       8.060000         8.060000   
456  8877689391   04-12-2016        2350       1.780000         1.780000   

     LoggedActivitiesDistance  VeryActiveDistance  ModeratelyActiveDistance  \
0                         0.0                2.57                      0.46   
1                         0.0                6.92                      0.73   
2                         0.0                4.66                      0.16   
3                         0.0                3.19                      0.79   
4                         0.0                2.16                      1.09   
..                        ...                 ...                       ...   
452                       0.0               11.10                      0.63   
453                       0.0                0.00                      0.02   
454                       0.0               21.92                      1.12   
455                       0.0                1.47                      0.15   
456                       0.0                0.00                      0.00   

     LightActiveDistance  SedentaryActiveDistance  VeryActiveMinutes  \
0                   4.07                     0.00                 33   
1                   3.91                     0.00                 89   
2                   3.71                     0.00                 56   
3                   4.95                     0.00                 39   
4                   4.61                     0.00                 28   
..                   ...                      ...                ...   
452                 8.62                     0.00                 70   
453                 8.02                     0.00                 90   
454                 4.46                     0.00                128   
455                 6.37                     0.01                 18   
456                 1.78                     0.00                  0   

     FairlyActiveMinutes  LightlyActiveMinutes  SedentaryMinutes  Calories  
0                     12                   205               804      1819  
1                     17                   274               588      2154  
2                      5                   268               605      1944  
3                     20                   224              1080      1932  
4                     28                   243               763      1886  
..                   ...                   ...               ...       ...  
452                   29                   359               982      4196  
453                    9                   289              1052      3841  
454                   46                   211              1055      4526  
455                    7                   225              1190      2820  
456                    0                    58               531       938  

[457 rows x 15 columns]

In [5]:
# 3. convert all the dates avaible in dataset to timestamp format in pandas 

In [6]:
df['ActivityDate'] = df['ActivityDate'].apply(lambda x: pd.Timestamp(x).strftime('%B-%d-%Y %I:%M %p'))

In [7]:
df.head(2)

Id            ActivityDate  TotalSteps  TotalDistance  \
0  1503960366  March-25-2016 12:00 AM       11004           7.11   
1  1503960366  March-26-2016 12:00 AM       17609          11.55   

   TrackerDistance  LoggedActivitiesDistance  VeryActiveDistance  \
0             7.11                       0.0                2.57   
1            11.55                       0.0                6.92   

   ModeratelyActiveDistance  LightActiveDistance  SedentaryActiveDistance  \
0                      0.46                 4.07                      0.0   
1                      0.73                 3.91                      0.0   

   VeryActiveMinutes  FairlyActiveMinutes  LightlyActiveMinutes  \
0                 33                   12                   205   
1                 89                   17                   274   

   SedentaryMinutes  Calories  
0               804      1819  
1               588      2154

In [8]:
# 4. Find out in this data that how many unique id's we have

In [9]:
array1 = df['Id'].unique()

In [10]:
array1  # It gives the list of unique ids

array([1503960366, 1624580081, 1644430081, 1844505072, 1927972279,
       2022484408, 2026352035, 2320127002, 2347167796, 2873212765,
       2891001357, 3372868164, 3977333714, 4020332650, 4057192912,
       4319703577, 4388161847, 4445114986, 4558609924, 4702921684,
       5553957443, 5577150313, 6117666160, 6290855005, 6391747486,
       6775888955, 6962181067, 7007744171, 7086361926, 8053475328,
       8253242879, 8378563200, 8583815059, 8792009665, 8877689391],
      dtype=int64)

In [11]:
len(array1)  # This gives the total no of unique ids in an array

35

In [12]:
# 5. which id is one of the active id that you have in whole dataset

In [13]:
df['Id'].max()    # It gives the maximum occurance of the id (active id)

8877689391

In [14]:
# 6 . how many of them have not logged there activity find out in terms of number of ids

In [15]:
df5 = df.loc[df['LoggedActivitiesDistance'] == 0]

In [16]:
df5.count()

Id                          433
ActivityDate                433
TotalSteps                  433
TotalDistance               433
TrackerDistance             433
LoggedActivitiesDistance    433
VeryActiveDistance          433
ModeratelyActiveDistance    433
LightActiveDistance         433
SedentaryActiveDistance     433
VeryActiveMinutes           433
FairlyActiveMinutes         433
LightlyActiveMinutes        433
SedentaryMinutes            433
Calories                    433
dtype: int64

433 ids have not logged their activity in the LoggedActivitiesDistance column

In [17]:
# 7. Find out who is the laziest person id that we have in dataset

In [18]:
df2 = df.groupby('Id',as_index=False)['SedentaryMinutes'].sum()

In [19]:
df2['SedentaryMinutes'].max()

43458

In [20]:
df2[df2['SedentaryMinutes'] == df2['SedentaryMinutes'].max()]

Id  SedentaryMinutes
14  4057192912             43458

Id 4.057193e+09	 is the most laziest as this id has most sedentary minutes (43458.0)

In [21]:
# 8. xplore over an internet that how much calories burn is required for a 
# healthy person and find out how many healthy person we have in our dataset

In [62]:
# From the internet, i have taken daily burn of calories between 2000 to 2500 is healthy.
# So, in the query i have given the range between 2000 to 2500 calories

In [63]:
df6 = df[(df['Calories']>=2000) & (df['Calories'] <= 2500)]

In [64]:
df6

Id            ActivityDate  TotalSteps  TotalDistance  \
1    1503960366  March-26-2016 12:00 AM       17609          11.55   
10   1503960366  April-04-2016 12:00 AM       14557           9.80   
11   1503960366  April-05-2016 12:00 AM       14844           9.73   
41   1644430081  April-04-2016 12:00 AM        2819           2.05   
62   1927972279  April-03-2016 12:00 AM        3198           2.21   
..          ...                     ...         ...            ...   
433  8792009665  April-01-2016 12:00 AM        4592           2.94   
436  8792009665  April-04-2016 12:00 AM        3821           2.45   
437  8792009665  April-05-2016 12:00 AM        2332           1.49   
438  8792009665  April-06-2016 12:00 AM        2121           1.36   
442  8792009665  April-10-2016 12:00 AM        4605           2.95   

     TrackerDistance  LoggedActivitiesDistance  VeryActiveDistance  \
1              11.55                       0.0                6.92   
10              9.80                       0.0                3.39   
11              9.73                       0.0                2.94   
41              2.05                       0.0                0.31   
62              2.21                       0.0                0.00   
..               ...                       ...                 ...   
433             2.94                       0.0                0.10   
436             2.45                       0.0                0.00   
437             1.49                       0.0                0.01   
438             1.36                       0.0                0.00   
442             2.95                       0.0                0.00   

     ModeratelyActiveDistance  LightActiveDistance  SedentaryActiveDistance  \
1                        0.73                 3.91                     0.00   
10                       0.70                 5.69                     0.00   
11                       0.76                 6.04                     0.00   
41                       0.17                 1.56                     0.01   
62                       0.00                 2.21                     0.00   
..                        ...                  ...                      ...   
433                      0.14                 2.70                     0.00   
436                      0.22                 2.23                     0.00   
437                      0.18                 1.30                     0.00   
438                      0.00                 1.36                     0.00   
442                      0.00                 2.95                     0.00   

     VeryActiveMinutes  FairlyActiveMinutes  LightlyActiveMinutes  \
1                   89                   17                   274   
10                  43                   18                   285   
11                  36                   18                   341   
41                   4                    4                    87   
62                   0                    0                   146   
..                 ...                  ...                   ...   
433                  4                    8                   176   
436                  0                    9                   161   
437                  1                   10                   111   
438                  0                    0                   122   
442                  0                    0                   179   

     SedentaryMinutes  Calories  
1                 588      2154  
10                757      2030  
11                736      2083  
41               1345      2444  
62               1183      2449  
..                ...       ...  
433               748      2260  
436               675      2229  
437               904      2100  
438               855      2114  
442              1261      2253  

[121 rows x 15 columns]

121 people spend daily calories in the range of 2000 to 2500. They are healthy

In [24]:
# 9. how many person are not a regular person with respect to activity try to find out those

In [66]:
# By regular, those who have more attendence to the gym
# people who are least regular have least attendence to the gym

In [67]:
df15 = df.groupby('Id',as_index=False)['ActivityDate'].count()

In [68]:
df15.sort_values('ActivityDate')

Id  ActivityDate
32  8583815059             8
10  2891001357             8
16  4388161847             8
25  6775888955             9
24  6391747486             9
2   1644430081            10
23  6290855005            10
22  6117666160            10
11  3372868164            10
21  5577150313            11
29  8053475328            11
34  8877689391            12
20  5553957443            12
28  7086361926            12
18  4558609924            12
33  8792009665            12
30  8253242879            12
12  3977333714            12
9   2873212765            12
31  8378563200            12
7   2320127002            12
6   2026352035            12
5   2022484408            12
4   1927972279            12
3   1844505072            12
15  4319703577            12
27  7007744171            12
26  6962181067            14
17  4445114986            15
8   2347167796            15
19  4702921684            15
1   1624580081            19
0   1503960366            19
14  4057192912            32
13  4020332650            32

The above query gives ids and their total count from the Activity dates (that is, how many days they attended the gym.)
It start from the lowest attendence of 8 days to the highest of 32 days. From this we can judge who is more regular to gym,
who is least regular to gym, who is average regular to the gym.

In [69]:
# 10 . who is the third most active person in this dataset find out those in pandas and in sql both

In [70]:
df11 = df.VeryActiveMinutes.nlargest(3).iloc[-1]

In [71]:
df11

128

In [72]:
df[df['VeryActiveMinutes'] == df.VeryActiveMinutes.nlargest(3).iloc[-1]]

Id            ActivityDate  TotalSteps  TotalDistance  \
454  8877689391  April-10-2016 12:00 AM       28497      27.530001   

     TrackerDistance  LoggedActivitiesDistance  VeryActiveDistance  \
454        27.530001                       0.0               21.92   

     ModeratelyActiveDistance  LightActiveDistance  SedentaryActiveDistance  \
454                      1.12                 4.46                      0.0   

     VeryActiveMinutes  FairlyActiveMinutes  LightlyActiveMinutes  \
454                128                   46                   211   

     SedentaryMinutes  Calories  
454              1055      4526

8877689391 is the third most active person in this dataset

In [73]:
# 11 . who is the 5th most laziest person avilable in dataset find it out

In [74]:
df10 = df.groupby('Id',as_index=False)['SedentaryMinutes'].sum()

In [75]:
df10.SedentaryMinutes.nlargest(5).iloc[-1]

15387

In [76]:
df10[df10['SedentaryMinutes'] == df10.SedentaryMinutes.nlargest(5).iloc[-1]]

Id  SedentaryMinutes
0  1503960366             15387

1503960366 is 5th most laziest person, who has total 15387 SedentaryMinutes

In [77]:
# 12 . what is a total acumulative calories burn for a person find out

In [78]:
df13 = df.groupby('Id',as_index=False)['Calories'].sum()

In [79]:
df13

Id  Calories
0   1503960366     34128
1   1624580081     25705
2   1644430081     29164
3   1844505072     19391
4   1927972279     27048
5   2022484408     29704
6   2026352035     16266
7   2320127002     18385
8   2347167796     30320
9   2873212765     20349
10  2891001357     18187
11  3372868164     18603
12  3977333714     16777
13  4020332650     98412
14  4057192912     60913
15  4319703577     23931
16  4388161847     14442
17  4445114986     31617
18  4558609924     21962
19  4702921684     42322
20  5553957443     21630
21  5577150313     36303
22  6117666160     20993
23  6290855005     21656
24  6391747486     15868
25  6775888955     24523
26  6962181067     29249
27  7007744171     31525
28  7086361926     26124
29  8053475328     31825
30  8253242879     17556
31  8378563200     40274
32  8583815059     19130
33  8792009665     24884
34  8877689391     41414